lib/utils/utils.sh

```diff
@@ -1,161 +1,185 @@
+    # shellcheck disable=SC1090
+    [ -f "$global_config" ] && source "$global_config"
+    local config=${BPKG_CONFIG:-"$HOME/.bpkgrc"}
+    # shellcheck disable=SC1090
+    [ -f "$config" ] && source "$config"
+    ## set defaults
+    if [ ${#BPKG_REMOTES[@]} -eq 0 ]; then
+      BPKG_REMOTES[0]=${BPKG_REMOTE-https://raw.githubusercontent.com}
+      BPKG_GIT_REMOTES[0]=${BPKG_GIT_REMOTE-https://github.com}
+
+      export BPKG_REMOTES
+      export BPKG_GIT_REMOTE
+      export BPKG_GIT_REMOTES
+    fi
 
-## Collection of shared bpkg functions
-
-## Init local config and set environmental defaults
-bpkg_initrc() {
-  local global_config=${BPKG_GLOBAL_CONFIG:-"/etc/bpkgrc"}
-  # shellcheck disable=SC1090
-  [ -f "$global_config" ] && source "$global_config"
-  local config=${BPKG_CONFIG:-"$HOME/.bpkgrc"}
-  # shellcheck disable=SC1090
-  [ -f "$config" ] && source "$config"
-  ## set defaults
-  if [ ${#BPKG_REMOTES[@]} -eq 0 ]; then
-    BPKG_REMOTES[0]=${BPKG_REMOTE-https://raw.githubusercontent.com}
-    BPKG_GIT_REMOTES[0]=${BPKG_GIT_REMOTE-https://github.com}
-
-    export BPKG_REMOTES
-    export BPKG_GIT_REMOTE
-    export BPKG_GIT_REMOTES
-  fi
-
-  export BPKG_PACKAGE_USER="${BPKG_PACKAGE_USER:-"bpkg"}"
-  export BPKG_INDEX=${BPKG_INDEX:-"$HOME/.bpkg/index"}
-
-  bpkg_validate
-
-  return $?
-}
-
-## check parameter consistency
-bpkg_validate () {
-  if [ ${#BPKG_GIT_REMOTES[@]} -ne ${#BPKG_REMOTES[@]} ]; then
-    bpkg_error "$(printf 'BPKG_GIT_REMOTES[%d] differs in size from BPKG_REMOTES[%d] array' "${#BPKG_GIT_REMOTES[@]}" "${#BPKG_REMOTES[@]}")"
-    return 1
-  fi
-  return 0
-}
-
-## format and output message
-bpkg_message () {
-  if type -f bpkg-term > /dev/null 2>&1; then
-    bpkg-term color "${1}"
-  fi
-
-  shift
-  echo -n "    ${1}"
-  shift
-
-  if type -f bpkg-term > /dev/null 2>&1; then
-    bpkg-term reset
-  fi
-
-  printf ": "
-
-  if type -f bpkg-term > /dev/null 2>&1; then
-    bpkg-term reset
-    bpkg-term bright
-  fi
-
-  printf "%s\n" "${@}"
-
-  if type -f bpkg-term > /dev/null 2>&1; then
-    bpkg-term reset
-  fi
-}
-
-## output error
-bpkg_error () {
-  {
-    bpkg_message "red" "error" "${@}"
-  } >&2
-}
-
-## output warning
-bpkg_warn () {
-  {
-    bpkg_message "yellow" "warn" "${@}"
-  } >&2
-}
-
-## output info
-bpkg_info () {
-  local title="info"
-  if (( "${#}" > 1 )); then
-    title="${1}"
+    export BPKG_PACKAGE_USER="${BPKG_PACKAGE_USER:-"bpkg"}"
+    export BPKG_INDEX=${BPKG_INDEX:-"$HOME/.bpkg/index"}
+
+    bpkg_validate
+
+    return $?
+  }
+
+  ## check parameter consistency
+  bpkg_validate () {
+    if [ ${#BPKG_GIT_REMOTES[@]} -ne ${#BPKG_REMOTES[@]} ]; then
+      bpkg_error "$(printf 'BPKG_GIT_REMOTES[%d] differs in size from BPKG_REMOTES[%d] array' "${#BPKG_GIT_REMOTES[@]}" "${#BPKG_REMOTES[@]}")"
+      return 1
+    fi
+    return 0
+  }
+
+
+  ## format and output message
+  bpkg_message () {
+    bpkg_exec_exist bpkg-term &&
```

Comment:
This is triggering something in the back of my mind but I can't exactly put my finger on it...

Something about using `&&` and not utilizing `\`, but I am not sure whether it is anything relevant. 

Thought it best to mention, though, just in case.